tree/treeplayer/src/TCsvDS.cxx

```diff
@@ -277,17 +276,14 @@ TCsvDS::TCsvDS(std::string_view fileName, bool readHeaders, char delimiter) // T
       // Infer types of columns with first record
       InferColTypes(columns);
 
-      // Read all records and store them in memory
-      do {
-         fRecords.emplace_back();
-         FillRecord(line, fRecords.back());
-      } while (std::getline(stream, line));
+      // Fill with the content of the first line
+      fRecords.emplace_back();
+      FillRecord(line, fRecords.back());
```

Comment:
just an idea, as this pattern is used below as well and is not super obvious -- we could do this instead:

```c++
FillRecord(line, std::back_inserter(fRecords));
```

Reply:
What is the advantage of this way? Left the old way for now.

Reply:
It's just that it took me a little bit to understand what was going on there, while with a `back_inserter` iterator it would be obvious that `FillRecord` adds a new record there. But it's not important.
There is also a performance argument, but it doesn't really apply in this case.